content/en/docs/rosenpass-tool/guides/ubuntu.md

```diff
@@ -0,0 +1,275 @@
+## Set up a Rosenpass-enhanced WireGuard VPN
+{{< /blocks/lead >}}
+
+{{% blocks/section color="light" class="no-flex contains-code-snippets compilation" %}}
+
+In this section, we set up a Rosenpass-enhanced WireGuard connection between two peers. Technically, there's no difference between the two peers. However, for clarity, we name them <span class="primary-highlight"><strong>server</strong> (pink)</span> and <span class="secondary-highlight"><strong>client</strong> (orange)</span>.
+
+### Prepare the Key Pairs
+
+#### 1. Start by generating secret keys for both peers
+
+**Note:** These will be stored in newly created `server.rosenpass-secret` and `client.rosenpass-secret` directories.
+
+```sh{class="starter-code-server"}
+rp genkey server.rosenpass-secret
+```
+```sh{class="starter-code-client"}
+rp genkey client.rosenpass-secret
+```
+
+#### 2. Extract the public keys
+
+**Note:** As above, these will be stored in newly created `server.rosenpass-public` and `client.rosenpass-public directories`.
+
+```sh{class="starter-code-server"}
+rp pubkey server.rosenpass-secret server.rosenpass-public
+```
+```sh{class="starter-code-client"}
+rp pubkey client.rosenpass-secret client.rosenpass-public
+```
+
+#### 3. Copy each `-public` directory to the other peer
+
+```sh{class="starter-code-server"}
+scp -r server.rosenpass-public user@client:/path/to/directory
+```
+```sh{class="starter-code-client"}
+scp -r client.rosenpass-public user@server:/path/to/directory
+```
+
+**Congrats! This completes the setup of the key pairs.**
+
+
+### Launch your Rosenpass-enhanced WireGuard VPN
+
+#### 4. Start the VPN
+
+**Note:** This may conflict with your firewall. In that case, you will need to configure your firewall to give Rosenpass access to the port number explicitly mentioned in these commands, `9999` in this example, as well as give WireGuard access to this port number incremented by one, `10000` in this example.
+
+```sh{class="starter-code-server"}
+sudo rp exchange server.rosenpass-secret dev rosenpass0 listen 127.0.0.1:9999 \
+peer client.rosenpass-public allowed-ips fe80::/64
+```
+```sh{class="starter-code-client"}
+sudo rp exchange client.rosenpass-secret dev rosenpass0 \
+peer server.rosenpass-public endpoint 127.0.0.1:9999 allowed-ips fe80::/64
+```
+
+#### 5. Assign IP addresses
+
+```sh{class="starter-code-server"}
+sudo ip a add fe80::1/64 dev rosenpass0
+```
+```sh{class="starter-code-client"}
+sudo ip a add fe80::2/64 dev rosenpass0
+```
+
+### Just to be sure: Verify the magic!
+
+#### 6. Test the connection
+
+You can test the connection by pinging the server from the client peer and vice versa:
+
+```sh{class="starter-code-client"}
+ping fe80::1%rosenpass0
+```
+```sh{class="starter-code-server"}
+ping fe80::2%rosenpass0
+```
+
+#### 7. Watch how Rosenpass replaces the WireGuard PSK
+
+You can watch how Rosenpass replaces the WireGuard PSK with the following command:
+
+```sh
+watch -n 0.2 'wg show all; wg show all preshared-keys'
```

Comment:
I'd just drop the `-n 0.2`, we switch keys every 120 seconds no need to refresh 5 times per second

Reply:
The `0.2` was from an old version, but regardlessly, I removed the current `-n 2`, because 2 seconds is the default anyway on Ubuntu.